File: Javascript/set.js
function TSet() {
    let items = {};

    this.has = function (value) {
        return items.hasOwnProperty(value);
    }
    this.add = function (value) {
        if (!this.has(value)) {
            items[value] = value;
            return true;
        }
        return false;
    }
    this.delete = function (value) {
        if (this.has(value)) {
            delete items[value];
            return true;
        }
        return false;
    }
    this.clear = function () {
        items = {};
    }
    this.size = function () {
        return Object.keys(items).length;
    }
    this.values = function () {
        let values = [];
        for (let i = 0, keys = Object.keys(items); i < keys.length; i++) {
            values.push(items[keys[i]]);
        }
        return values;
    }
    this.union = function (otherSet) {
        let unionSet = new TSet();
        let values = this.values();
        for (let i = 0; i < values.length; i++) {
            unionSet.add(values[i]);
        }
        values = otherSet.values();
        for (let i = 0; i < values.length; i++) {
            unionSet.add(values[i]);
        }
        return unionSet;
    }
    this.intersection = function (otherSet) {
        let intersectionSet = new TSet();
        let values = this.values();
        for (let i = 0; i < values.length; i++) {
            if (otherSet.has(values[i])) {
                intersectionSet.add(values[i]);
            }
        }
        return intersectionSet;
    }
    this.difference = function (otherSet) {
        let differenceSet = new TSet();
        let values = this.values();
        for (let i = 0; i < values.length; i++) {
            if (!otherSet.has(values[i])) {
                differenceSet.add(values[i]);
            }
        }
        return differenceSet;
    }
    this.subset = function (otherSet) {
        if (this.size() > otherSet.size()) {
            return false;
        }
        else {
            let values = this.values();
            for (let i = 0; i < values.length; i++) {
                if (!otherSet.has(values[i])) {
                    return false;
                }
            }
            return true;
        }5
    }
}
let set = new TSet();
set.add('test');
set.add(7);
set.add('js');
console.log(set.size());
console.log(set.has(8));
console.log(set.has('test'));
console.log(set.delete(7));
console.log(set.delete(6));
console.log(set.has(7));
console.log(set.values());

let setA = new TSet(); setA.add(1); setA.add(2); setA.add(3); setA.add(9);
let setB = new TSet(); setB.add(3); setB.add(4); setB.add(2); setB.add(6);
let setC = new TSet(); setC.add(10); setC.add(1); setC.add(2); setC.add(3); setC.add(9);

let unionAB = setA.union(setB);
console.log(unionAB.values());

let intersectionAB = setA.intersection(setB);
console.log(intersectionAB.values());

let differenceAB = setA.difference(setB);
console.log(differenceAB.values());

console.log(setA.subset(setB));
console.log(setA.subset(setC));

File: Python/set.py
class Set():
    def __init__(self):
        self.items = {}
    
    def has(self, value):
        if self.items.get(value):
            return True
        return False
    
    def add(self, value):
        if not self.has(value):
            self.items[value] = value
            return True
        return False
    
    def delete(self, value):
        if self.has(value):
            self.items.pop(value)
            return True
        return False

    def clear(self):
        self.items = {}

    def size(self):
        return len(self.items)
    
    def values(self):
        values = []
        for k in self.items:
            values.append(self.items.get(k))
        return values
    
    def union(self, otherSet):
        unionSet = Set()
        values = self.values()
        for v in values:
             unionSet.add(v)
        values = otherSet.values()
        for v in values:
            unionSet.add(v)
        return unionSet
    
    def intersection(self, otherSet):
        intersectionSet = Set()
        values = self.values()
        for v in values:
            if otherSet.has(v):
                intersectionSet.add(v)
        return intersectionSet

    def difference(self, otherSet):
        differenceSet = Set()
        values = self.values()
        for v in values:
            if not otherSet.has(v):
                differenceSet.add(v)
        return differenceSet

    def subset(self, otherSet):
        if self.size() > otherSet.size():
            return False
        else:
            values = self.values()
            for v in values:
                if not otherSet.has(v):
                    return False
        return True
    
def main():
    set = Set()
    set.add(2)
    set.add(2)
    set.add(9)
    print(set.has(2))
    set.add(6)
    set.add(4)
    print(set.size())
    print(set.has(3))
    set.add(9)
    print(set.items)
    set.delete(4)
    print(set.items)
    set.delete(2)
    print(set.items)
    print(set.size())
    print(set.values())
    set.add(13)
    set.add(49)
    print(set.items)
    print(set.size())
    print(set.values())

    setA = Set()
    setA.add(1)
    setA.add(2)

    setB = Set()
    setB.add(3)
    setB.add(4)

    setC = Set()
    setC.add(2)
    setC.add(9)
    setC.add(1)
    setC.add(3)

    unionAB = setA.union(setB)
    print(unionAB.values())

    intersectionAC = setA.intersection(setC)
    print(intersectionAC.values())
    
    differenceCB = setC.difference(setB)
    print(differenceCB.values())

    print(setA.subset(setB))
    print(setA.subset(setC))

if __name__ == "__main__":
    main()